### PR TITLE
feat(pgwire): listen on unix socket

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Allow listening on a unix socket for pgwire. Not much useful now but it's the default behavior of the Postgres server.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by TennyZhuang — APPROVED** [review]

File: src/utils/pgwire/src/pg_server.rs [hunk 1]
```diff
@@ -130,53 +128,99 @@ impl UserAuthenticator {
...
+#[auto_enums::enum_derive(tokio1::AsyncRead, tokio1::AsyncWrite)]
+enum Stream {
+    Tcp(TcpStream),
+    Unix(UnixStream),
+}
+
+impl Listener {
+    async fn bind(addr: &str) -> io::Result<Self> {
+        if addr.starts_with("unix:") {
+            let path = &addr[5..];
```
Comment (wangrunji0408) [inline src/utils/pgwire/src/pg_server.rs:145]:
```suggestion
        if let Some(path) = addr.strip_prefix("unix:") {
```

File: src/utils/pgwire/src/pg_server.rs [hunk 2]
```diff
@@ -130,53 +128,99 @@ impl UserAuthenticator {
...
+            Self::Unix(listener) => {
+                let (stream, addr) = listener.accept().await?;
+
+                Ok((Stream::Unix(stream), format!("{addr:?}")))
+            }
+        }
+    }
+}
+
+/// Binds a Tcp or Unox listener at `addr`. Spawn a coroutine to serve every new connection.
```
Comment (wangrunji0408) [inline src/utils/pgwire/src/pg_server.rs:169]:
```suggestion
/// Binds a Tcp or Unix listener at `addr`. Spawn a coroutine to serve every new connection.
```

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

File: src/utils/pgwire/src/pg_server.rs (hunk 1 above)
Comment (BugenZhao) [reply src/utils/pgwire/src/pg_server.rs:145]: This is cool, but the cooler thing is that clippy can lint on this. 🤯

**Pull request merged**